monix/shared/src/main/scala/org/atnos/eff/monix/AsyncTaskService.scala

```diff
@@ -40,7 +40,7 @@ case class AsyncTaskService() extends AsyncService {
 trait AsyncTaskServiceInterpretation {
 
   def runAsyncTask[A](e: Eff[Fx.fx1[Async], A]): Task[A] =
-    e.detachA(ApplicativeAsync) match {
```

Comment:
Ohhhh my, I see. The monad is getting picked by implicit resolution. Gosh darn it :P